src/main/java/teammates/ui/webapi/output/SearchCoursesResult.java

```diff
@@ -0,0 +1,25 @@
+package teammates.ui.webapi.output;
+
+import java.util.List;
+
+/**
+ * Contains Course search results.
+ */
+public class SearchCoursesResult extends ApiOutput {
+    private List<SearchCourseData> students;
+    private List<SearchCourseData> instructors;
```

Comment:
Is it necessary to return a object containing `students ` and `instructors `. It seems that we can do this **uniformly** (REST Principle) by just returning a list of `courseData`

Reply:
I think this was done so we could skip one filter operation on the front end side when joining the data

Reply:
But according to your doc, when returning, one field will be empty and the other will be of content right? Therefore, the frontend can determine what the result is for (`students` or `instructor`)?

A more strong reason is, the endpoint is not **uniform** and is not designed with a fine granularity.

Reply:
Sure.